helix-term/src/ui/mod.rs

```diff
@@ -283,14 +283,26 @@ pub mod completers {
         names
     }
 
+    /// Recursive function to get all keys from this value and add them to vec
+    fn get_keys(value: &serde_json::Value, vec: &mut Vec<String>, scope: Option<&str>) {
+        if let Some(map) = value.as_object() {
+            for (key, value) in map.iter() {
+                let key = match scope {
+                    Some(scope) => format!("{}.{}", scope, key),
+                    None => key.clone(),
+                };
+                get_keys(value, vec, Some(&key));
+                vec.push(key);
```

Comment:
```suggestion
                if !value.is_object() {
                    vec.push(key);
                }
```

I think we should only show completions for the leaves of the object trees here. For example we should show `whitespace.render` but not `whitespace`